src/Core/Domain/Customer/Query/SearchCustomers.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Customer\Query;
+
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerException;
+
+/**
+ * Searchers for customers by phrases matching customer's first name, last name, email and id
+ */
+class SearchCustomers
+{
+    /**
+     * @var string[]
+     */
+    private $phrases;
```

Comment:
@matks i did some changes regarding your comments and introduced `string[] $phrases` instead of `string $query`. Basically, if you pass phrases to Query, it will perform search against customer id, first name, last name and email. And this covers our current needs. 

However, you can image having `string[] $firstNames` or `string[] $emails` properties in `SearchCustomers` Query in the future in case we have a need to just perform search by emails (or first names and etc).

wdyt? im not sure if that's what you meant. :thinking:

Reply:
👍 I think that is a very good answer to my concerns 🎉